### ISSUE TITLE
Compiler warnings when compiling vsClojure

### ISSUE THREAD

**Issue by frankhale** [vsClojure/vsClojure#101]
I've followed the instructions on the following page to resolve the reference conflicts but it doesn't seem to solve the problem, in fact it compounds the problem and produces more of the same types of warnings.

http://msdn.microsoft.com/en-us/library/ff182188.aspx

A reference was created to embedded interop assembly 'c:\Program Files (x86)\Common Files\Microsoft Shared\MSEnv\PublicAssemblies\envdte.dll' because of an indirect reference to that assembly created by assembly 'c:\Users\fhale\Desktop\vsClojure\Dependencies\Visual Studio SDK\Microsoft.VisualStudio.Shell.10.0.dll'. Consider changing the 'Embed Interop Types' property on either assembly.

![embeded-assembly-warning](https://f.cloud.github.com/assets/190154/1734490/e874686a-6346-11e3-8c28-8fac873d5e31.PNG)

**Comment by speige** [comment]
I've tried to fix these as well & couldn't figure out how. There doesn't seem to be any problem at runtime caused by the warnings.

**Comment by frankhale** [comment]
Yeah I haven't ran into any problems, just wanted to see if I could make
those go away. I'm still ramping up on VS SDK programming so it may take me
a little while to get productive.

**Comment by frankhale** [comment]
I'm labeling this a bug even though it doesn't cause issues. I'm going to continue to research this to find a solution that works.

**Comment by frankhale** [comment]
This one still seems tricky. I did find this reference on MSDN:

http://msdn.microsoft.com/en-us/library/envdte.aspx

```
EnvDTE is an assembly-wrapped COM library containing the objects and members for Visual Studio core automation.
Changes and new functionality are contained in the EnvDTE80, EnvDTE90, EnvDTE90a, and EnvDTE100 namespaces.
When you add an assembly reference to EnvDTE.dll, you must also set the Embed Interop Types property of the assembly to false.
```

Notice it says when you add a reference to EnvDTE you must set the Embed Interop Types prop to false. I can safely say this has not been done in at least one place (ClojureExtension.Repl). However, it's not so easy to just flip these to false. We have references to the sub assemblies EnvDTE80, et al. and this is part of the reason it is complaining. Removing references is also not a solution so right now I'm at a loss for how to resolve this craziness. I'm going to continue to research this.

**Comment by speige** [comment]
I think the issue was that an assembly (EnvDTE) needs to have the same setting for "embed interop types" for all references, to that dll from all projects in the solution. I tried to do that, but had trouble with warnings or errors no matter which way I tried it.

However, I found that the only compiler error was a missing "Text" property on dte.ActiveDocument.Selection. With "embed interop types" true, I saw that the Selection property was being compiled as type "dynamic", but with the "embed interop types" false, it was being compiled as type "object". I casted it to dynamic in code & set "embed interop types" to false for all dlls in the solution. All warnings are gone & the code works correctly at runtime.

I found this link http://blogs.clariusconsulting.net/kzu/check-your-embed-interop-types-flag-when-doing-visual-studio-extensibility-work/ which explains the purpose being embed interop & that it's not necessary, & specifically mentions that it causes major painpoints while doing visual studio extensions development.
